scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala

```diff
@@ -229,9 +230,11 @@ private[types] object TypeProvider {
 
   /** Generate a case class. */
   private def caseClass(c: blackbox.Context)
-                       (name: c.TypeName, fields: Seq[c.Tree], body: Seq[c.Tree]): c.Tree = {
+                       (mods: c.Modifiers, name: c.TypeName, fields: Seq[c.Tree], body: Seq[c.Tree]): c.Tree = {
     import c.universe._
-    q"@_root_.com.spotify.scio.bigquery.types.BigQueryTag case class $name(..$fields) extends ${p(c, SType)}.HasAnnotation { ..$body }"
+    val tagAnnot = q"new _root_.com.spotify.scio.bigquery.types.BigQueryTag"
+    val taggedMods = Modifiers(Flag.CASE, typeNames.EMPTY, tagAnnot :: mods.annotations)
+    q"$taggedMods class $name(..$fields) extends ${p(c, SType)}.HasAnnotation { ..$body }"
```

Comment:
do we need the `case` modifier or is that what the first argument to this function does